pkg/module/go_get_fetcher.go

```diff
@@ -0,0 +1,163 @@
+package module
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	pkgerrors "github.com/pkg/errors"
+	"github.com/spf13/afero"
+)
+
+var (
+	// ErrLimitExceeded signals that github.com refused to serve the request due to exceeded quota
+	ErrLimitExceeded = errors.New("github limit exceeded")
+)
+
+type goGetFetcher struct {
+	fs           afero.Fs
+	repoURI      string
+	version      string
+	goBinaryName string
+}
+
+// NewGoGetFetcher creates fetcher which uses go get tool to fetch modules
+func NewGoGetFetcher(goBinaryName string, fs afero.Fs, repoURI, version string) (Fetcher, error) {
+	if repoURI == "" {
+		return nil, errors.New("invalid repository identifier")
+	}
+
+	return &goGetFetcher{
+		fs:           fs,
+		repoURI:      repoURI,
+		version:      version,
+		goBinaryName: goBinaryName,
+	}, nil
+}
+
+// Fetch downloads the sources and returns path where it can be found. This function will
+// always return a ref, even if it returns a non-nil error.
+//
+// If an error was returned, the returned ref's Read method will return an error, but you
+// should always call ref.Clear() on the returned ref
+func (g *goGetFetcher) Fetch(mod, ver string) (Ref, error) {
```

Comment:
so why do we always wanna return a Ref? why can't we just return `nil`? If it's a matter of calling `ref.Clear()` we can always make sure we clear things inside this function when an error occurs, no?

Reply:
>also, why does Fetch takes a mod/ver when NewGoGetFetcher also takes a mod and ver?

good catch, only `Fetch` should take the mod/ver